packaging/opensuse/snapd.spec

```diff
@@ -168,83 +200,67 @@ export CFLAGS
 export CXXFLAGS
 export LDFLAGS
 
-# N.B.: Prior to openSUSE Tumbleweed in May 2018, the AppArmor userspace in SUSE
-# did not support what we needed to be able to turn on basic integration.
+# Generate autotools build system files.
+pushd %{indigo_srcdir}/cmd
+autoreconf -i -f
 %configure \
     %{!?with_apparmor:--disable-apparmor} \
     --libexecdir=%{_libexecdir}/snapd \
     --enable-nvidia-biarch \
     %{?with_multilib:--with-32bit-libdir=%{_prefix}/lib} \
     --with-snap-mount-dir=%{snap_mount_dir} \
     --enable-merged-usr
+popd
 
 %build
-# Build golang executables
-%goprep %{import_path}
-
+# Build golang executables, with the following exceptions everything is built the same way:
+# - snap-exec and snap-update-ns is built statically.
+# - snapd has a variant that uses test keys instead of production keys.
+#
+# NOTE: indigo_gopath takes priority over GOPATH. This ensures that we
+# build the code that we intended in case GOPATH points to another copy.
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snap
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snapctl
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=pie %{import_path}/cmd/snap-seccomp
+GOPATH=%{indigo_gopath}:$GOPATH go build -buildmode=default -ldflags '-extldflags "-static"' %{import_path}/cmd/snap-update-ns
```

Comment:
I think this should include distro `%_ldflags` in `-extldflags` (probably for all the binaries we build). I know Fedora is quite strict about that, not sure about openSUSE though.

Reply:
`rpm -E '%{_ldflags}'` doesn't expand here, perhaps it exits just for fedora?

Reply:
Fedora has `redhat-rpm-config` which sets that, but openSUSE does not seem to have anything similar. Wonder whether they actually use any flags enforced by distro policy, maybe not.

Reply:
They must have something but it applies to C code via a mechanism I'm not familiar with. When doing initial review of the package submission to factory (long time ago) we got flagged on this and fixed it.